### PR TITLE
Breakable Window API + Add DamageTypeExtensions Method

### PR DESCRIPTION
**Pull request by louis1706**
- Window API
- Exiled.API.Enum : ``GlassType``
- Add ``GetDamageType(DamageHandlerBase )``
- Make public ``TranslationConversion`` and add ``TranslationIdConversion``
- New Event ``Handlers.Player.PlayerDamageWindow``

~~(Not tested for now)~~

### CODE REVIEW

File: Exiled.API/Features/Window.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        /// Gets or sets the window's rotation.
+        /// </summary>
+        public Quaternion Rotation
+        {
+            get => GameObject.transform.rotation;
+            set => GameObject.transform.rotation = value;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not this window is broken.
```
Comment (Thundermaker300) [inline Exiled.API/Features/Window.cs:109]: doc

File: Exiled.API/Features/Window.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        /// <returns>A <see cref="Door"/> wrapper object.</returns>
+        public static Window Get(BreakableWindow breakableWindow) => BreakableWindowToWindow.ContainsKey(breakableWindow)
+            ? BreakableWindowToWindow[breakableWindow]
+            : new Window(breakableWindow);
+
+        /// <summary>
+        /// Damages the window.
+        /// </summary>
+        /// <param name="amount">The amount of damage to deal.</param>
+        public void DamageDoor(float amount) => Base.ServerDamageWindow(amount);
```
Comment (Thundermaker300) [inline Exiled.API/Features/Window.cs:148]: Rename to DamageWindow or Damage, not DamageDoor

**Comment by joker-119** [comment]
Your DamageHandler conversion is unnecessary.. literally just ev.Handler.Type, when will you ever have a DamageHandler that isn't in an event?

**Comment by louis1706** [comment]
> Your DamageHandler conversion is unnecessary.. literally just ev.Handler.Type, when will you ever have a DamageHandler that isn't in an event?

It's nessecary in an harmony patch you don't need to recreated a new DamageHandler for just get an player

File: Exiled.API/Features/Window.cs (hunk 1 above)
Comment (louis1706) [reply Exiled.API/Features/Window.cs:109]: ?

**Comment by louis1706** [comment]
All the Window with ToTring() for testing (work fine)
```
Plants 10 False False
Plants 10 False False
MicroHid 10 False False
Plants 10 False False
Scp330 120 False False
GR18 20 False False
Scp079 5 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
Scp079 5 False False
GR18 20 False False
Scp049 10 False False
GR18 20 False False
GR18 20 False False
GR18 20 False False
Scp079Trigger 5 False True```
I have tested the position and rotation too
And GetDamageType work
and i have verified for the event swith SanyaDevTools

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]
Fix build errors + Rename the event to `DamagingWindowEventArgs`.

**Comment by louis1706** [comment]
> Fix build errors + Rename the event to `DamagingWindowEventArgs`.

I do the rename later i don't going to do it on phone :)

**Pull request merged**